metagraph/src/graph/representation/succinct/dbg_succinct.hpp

```diff
@@ -53,13 +64,25 @@ class DBGSuccinct : public DeBruijnGraph {
                               const std::function<void(node_index)> &callback,
                               const std::function<bool()> &terminate = [](){ return false; }) const override;
 
+    // Traverse graph mapping sequence to the graph nodes
+    // and run callback for each node until the termination condition is satisfied.
+    // For each k-mer satisfying the skip condition, run the callback on npos.
+    // Guarantees that nodes are called in the same order as the input sequence.
+    // In canonical mode, non-canonical k-mers are NOT mapped to canonical ones
+    virtual void map_to_nodes_sequentially_checked(std::string_view sequence,
```

Comment:
Do you think it's worth adding this to `DeBruijnGraph` as well?

Reply:
Would be better to just add this skipper to all functions `map_to_nodes_sequentially`